background.js

```diff
@@ -25,3 +56,58 @@ function delay(i) {
     console.log(i);
   }, i * 1000);
 }
+
+chrome.tabs.onUpdated.addListener(function(tabId, info, tab) {
+    chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
+        const activeTab = tabs[0]
+        updateTabId(activeTab.id)
+        currentTranslatingPage = removeItemAll(currentTranslatingPage, previousTabID)
+        if (activeTab.url.includes("developer.apple.com")) {
+            if (shouldTranslate) {
+                if (currentTranslatingPage.includes(currentTabID)) {
+                    chrome.action.setIcon({ path: { "128": "/source/intro/swiftLogo-translating.png"} }).then(r => {})
+                } else {
+                    console.log(currentTranslatingPage, tab.id)
+                    chrome.action.setIcon({ path: { "128": "/source/intro/swiftLogo-running.png"} }).then(r => {})
+                }
+            } else {
+                chrome.action.setIcon({ path: { "128": "/source/intro/swiftLogo.png"} }).then(r => {})
+            }
+        } else {
+            chrome.action.setIcon({ path: { "128": "/source/intro/swiftLogo.png"} }).then(r => {})
+        }
+    });
+});
+
+chrome.tabs.onActivated.addListener(function(activeInfo) {
```

Comment:
is it necessary to handle both `onActivated` and `onUpdated` methods to update the icon?